packages/rs-dpp/src/data_contract/document_type/document_field.rs

```diff
@@ -281,71 +289,88 @@ impl DocumentFieldType {
         buf: &mut BufReader<&[u8]>,
         required: bool,
     ) -> Result<Option<Value>, ProtocolError> {
+        if !required {
+            let marker = buf.read_u8().map_err(|_| {
```

Comment:
I'm not sure what marker does. Is it just to signify that the value of that field is null?